src/lm_polygraph/generation_metrics/sbert.py

```diff
@@ -0,0 +1,49 @@
+import numpy as np
+from sentence_transformers import SentenceTransformer, util
+
+from typing import List, Dict
+from .generation_metric import GenerationMetric
+
+
+class SbertMetric(GenerationMetric):
+    def __init__(self):
+        super().__init__(['greedy_texts'], 'sequence')
+        self.sbert = SentenceTransformer("all-mpnet-base-v2")
+
+    def __str__(self):
+        return f'Sbert'
+
+    def _score_single(self, t1: str, t2: str):
+        return util.cos_sim(self.sbert.encode(t1), self.sbert.encode(t2)).item()
+
+    def __call__(self, stats: Dict[str, np.ndarray], target_texts: List[str],
+                 target_tokens: List[List[int]]) -> np.ndarray:
+        return np.array([self._score_single(hyp, ref) for hyp, ref in zip(stats['greedy_texts'], target_texts)])
```

Comment:
Can we do it in two batches? First score full batch of greedy texts, then full batch of targets, and and then cosine.